metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -82,12 +84,16 @@ func NewResourceMetadataEnricher(
 	res kubernetes.Resource,
 	nodeScope bool) Enricher {
 
-	config := ValidatedConfig(base)
-	if config == nil {
+	config, err := GetValidatedConfig(base)
+	if err != nil {
 		logp.Info("Kubernetes metricset enriching is disabled")
 		return &nilEnricher{}
 	}
 
+	if PerfMetrics.Timeout < config.CacheExpirationTime {
+		PerfMetrics = NewPerfMetricsCache(config.CacheExpirationTime)
```

Comment:
Will this overwrite the values if `NewResourceMetadataEnricher` is called more than once? Like at [state_container](https://github.com/elastic/beats/blob/89bcc33a9a90cbacceaa61fceec0e8d23073574c/metricbeat/module/kubernetes/state_container/state_container.go#L121) and [container](https://github.com/elastic/beats/blob/89bcc33a9a90cbacceaa61fceec0e8d23073574c/metricbeat/module/kubernetes/container/container.go#L78) ?

The result should be the same but you just initialise the cache twice and the second call to `NewContainerMetadataEnricher` overwrites the already existent value/object/reference.

A workaround here would be to initialise the cache globally like it was happening already and then use `SetExpirationTime()` method to set the expiration. This new method can be called multiple times but needs to ensure that only updates the timeout if the incoming value is bigger than the already set. In this was you ensure that you use the `max()` of the various scrape.periods.